### ISSUE TITLE
Is there a clear definition for what validation of a function class is to mean across tools?

### ISSUE THREAD

**Issue by gwr69** [modelica/ModelicaSpecification#2838]
In the wake of the discussion in #2835 I loaded my library in `OpenModelica v1.16.0-dev.03 (64-bit)` and validated all classes in order to improve compatibility across platforms.

I was surprised to find a report for quite a few `function` classes as having `failed` the validation check. For example, this simple class fails validation in Open Modelica:

```
function foo
    input Boolean[:] x;
    output Boolean y;
algorithm
    y := true;
end foo;
```

<s>Validating this class</s>Using `Check Model` in  `OpenModelica v1.16.2 (64-bit)` gives:

> [1] 11:32:47 Translation Error
> [foo: 2:3-2:21]: Failed to deduce dimension 1 of x due to missing binding equation.

Validating the same class in `Wolfram System Modeler 12.2` gives:

> [3] 10:59 Validation of function foo 
> Validation of function foo completed successfully.

This begs the question what validation of a class is supposed to mean for such a function class. Can you _fail_ a _test that could not be performed_?

**Comment by maltelenz** [comment]
In my opinion the language specification should not get involved in what "validation" might mean to deeply. It should specify what is correct in what context.

In this case, the function as such, stand-alone, is not invalid. Clearly, there are things about it that cannot be determined until it is used in a larger context (such as the dimension of `x`). This is fine, and tools might choose to mention this, or not, in different contexts (validation, running in some kind of REPL-like interface, used in a larger model, when editing, ...) as they see fit.

I propose this is a tool issue, and it would be to hard to agree on, or even come up with, all the various scenarios where a tool could say or do one thing or another.

@gwr69 Feel free to disagree or explain if you think this should be included in the standard. You as a user and library developer probably have a different background and insights.

**Comment by gwr69** [comment]
Ok, I have to correct myself a bit: While the UI buttons in Wolfram System Modeler and OMEdit look quite alike (e.g. check marks), the functionality in System Modeler is explicitly called `Validate Class` while the one in OMEdit is called `Check Model`.

So getting an `error` then is strictly _correct_—but _unfortunate_ from a user (or library developer) perspective. We would like to quickly check classes and in my case a library to spot Modelica issues pertaining to syntax and semantics the later probably being a difficult issue in equation based modeling.

TL;DR I would tend to agree with your point of view, Malte.

**Comment by HansOlsson** [comment]
I agree with @maltelenz that we shouldn't specify exactly what kinds of correctness checks are possible, and what different UI-elements in tools should do. So, it seems we can close this as tool-issue.

Note for completeness: the specification https://specification.modelica.org/master/introduction1.html#scope-of-the-specification includes some statements about validity of functions.
